kubernetes/cache/cache.go

```diff
@@ -145,7 +146,17 @@ func NewKialiCache(clientFactory kubernetes.ClientFactory, cfg config.Config) (K
 	}
 
 	for cluster, client := range clientFactory.GetSAClients() {
-		cache, err := NewKubeCache(client, cfg)
+		var accessibleNamespaces []string
+		if !cfg.Deployment.ClusterWideAccess {
+			// Determine what namespaces are to be accessed for the cluster.
+			// Since we do not have cluster wide access, we do not have permission to list namespaces.
+			// Therefore, we assume we can extract the list of accessible namespaces from the discovery selectors configuration.
+			// That list of accessible namespaces will be used as our base list which we then filter with discovery selectors later.
+			// Note if this is a remote cluster, that remote cluster must have the same namespaces as those in our own local
+			// cluster's accessible namespaces. This is one reason why we suggest enabling CWA for multi-cluster environments.
+			accessibleNamespaces = cfg.ExtractAccessibleNamespaceList()
```

Comment:
Should this be part of config validation?: https://github.com/kiali/kiali/blob/61e8b992bc8be877b53017a61983d866ec9d323a/config/config.go#L1195

That way if the list is malformed the server can fail fast rather than silently drop errors. You could extract once there and just save the list on the config object for other places to consume, similar to `cfg.Deployment.AccessibleNamespaces` except it shouldn't be a config opt that the user can set.

Reply:
Either way, since the list comes from the config object and we're already passing the config object into the cache constructor, do you need to add the `accessibleNamespaces` as a separate argument to the cache constructor or just pull it from the config inside `NewKubeCache`?

Reply:
This is now done. It is done as part of the unmarshalling; it isn't done during validation. There is no modification of the config in the validate function - that strictly looks at the config as it is and gives it a thumbs-up or a thumbs-down. The Unmarshal method is where additional work is done to massage the config, so i added this stuff there.